test/spmd/test_xla_sharding.py

```diff
@@ -50,9 +50,11 @@ def test_xla_shards(self):
       start, end = (i, i + 1) * shard_len
       expected = torch.arange(start, end, dtype=torch.float32)
       self.assertTrue(torch.allclose(shard.data, expected))
-      self.assertIsInstance(shard.indices, list)
-      self.assertEqual(len(shard.indices), len(t.shape))
-      self.assertEqual(shard.indices[0], slice(start, end, 1))
+      if isinstance(shard.indices, list):
+        self.assertEqual(len(shard.indices), len(t.shape))
+        self.assertEqual(shard.indices[0], slice(start, end, 1))
+      else:
+        self.assertIsInstance(shard.indices, type(Ellipsis))
```

Comment:
Ah sharding across one device is `MAXIMAL` which becomes `REPLICATED`, so this happens on CPU 👍